### ISSUE TITLE
"Failed to join room" against Synapse 1.12.0

### ISSUE THREAD

**Issue by cvincent** [kfatehi/matrix-puppet-imessage#40]
Last night I began the process of moving my homeserver off my media server to a new server dedicated to various cloud applications (so the media server can dedicate all its resources to transcoding). This meant getting my lovely iMessage bridge moved as well.

I ended up in a situation where I could receive but not send messages. After many frustrating hours of debugging, I realized my new homeserver installation was of course using the latest version (1.12.0 as of this writing), while the previous installation which had been working was on 1.9.1. So I rolled back to 1.9.1, started everything from scratch, and now I'm back in business.

Here's the error I see (phone number edited out) when running against 1.12.0: `failed to join ghost +XXXXXXXXXXX Error: Failed to join room`.

I don't know which precise Synapse version breaks it, only that 1.9.1 works and 1.12.0 doesn't. It's possible the issue isn't in the iMessage bridge itself, but in the underlying library it uses to connect with Matrix.

**Comment by cvincent** [comment]
Turns out this isn't totally unknown:
https://matrix.to/#/!ChuQQIVJvwyJujhNIG:synapse.keyvan.pw/$15844732593TBqfJ:bobpaul.org?via=matrix.org&via=tchncs.de&via=matrix.allmende.io

**Comment by nr23730** [comment]
Did you use the latest commit 1bb03853f18f36596c15a1c6b4476a6827557e96? I fixed the issue there...

**Comment by Philantrop** [comment]
> Did you use the latest commit [1bb0385](https://github.com/matrix-hacks/matrix-puppet-imessage/commit/1bb03853f18f36596c15a1c6b4476a6827557e96)? I fixed the issue there...

I'm using a checkout of that commit and am seeing the same symptoms as cvincent. Messages from Matrix are never sent to iMessage but the inverse way works just fine. It started with Synapse 1.12.0 for me, too. It worked with v1.11.0.

**Comment by cwmke** [comment]
Is this for new or existing conversations? I'm on 1.12.0 and haven't noticed any problems with existing conversations. I haven't had any new conversations to test with.

**Comment by Philantrop** [comment]
> Is this for new or existing conversations? I'm on 1.12.0 and haven't noticed any problems with existing conversations. I haven't had any new conversations to test with.

Both in my case.

**Comment by cvincent** [comment]
Yeah this is a separate issue from what was fixed in 1bb0385 (which is merging in my fix for the other issue).

**Comment by cvincent** [comment]
From a little sleuthing starting from the above matrix.to link, it appears to be related to this issue: https://github.com/matrix-org/matrix-doc/pull/2261

I'm wondering if simply updating the underlying matrix-puppet-bridge library will make it work against the latest Synapse version. Can anyone confirm? I'd try it but I do rely on my existing setup and don't want to end up in a weird situation that's hard to come back from. Would be great to get some kind of automated testing around bridge integration in general.
